### PR TITLE
[Feature] Add doc system based on Jekyll

### PR DESCRIPTION
**Pull request by frankliee**
### What changes were proposed in this pull request?

Add doc system based on Jekyll.

### Why are the changes needed?

Display document  (such as JavaDoc and system design) on web site.

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

cd docs
sh gen-doc.sh
jekyll serve --host 0.0.0.0 --port 4000

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -857,6 +857,7 @@
                 <exclude>quantity_temp.properties</exclude>
                 <exclude>rss*/</exclude>
                 <exclude>spark-patches/**</exclude>
+                <exclude>docs/*.md</exclude>
```
Comment (jerqi) [inline pom.xml:860]: why should docs be ecluded?

File: docs/pageA.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
```
Comment (jerqi) [inline docs/pageA.md:1]: what's the meaning of pageA?

Comment (frankliee) [reply docs/pageA.md:1]: Just an example.

File: docs/_config.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+title: Firestorm homepage
+baseurl: ""
+url: ""
+
+# Build settings
+theme: minima
+
+kramdown:
+  math_engine: mathjax
+  syntax_highlighter: rouge
```
Comment (duanmeng) [inline docs/_config.yml:28]: A a new line at the end of the file

File: docs/gen-doc.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations under the License.
+#
+
+set -e
+
+cd ..
```
Comment (duanmeng) [inline docs/gen-doc.sh:23]: use dirname command to get absolute path

File: pom.xml (hunk 1 above)
Comment (frankliee) [reply pom.xml:860]: Like Spark

File: docs/_config.yml (hunk 3 above)
Comment (frankliee) [reply docs/_config.yml:28]: ok

File: docs/gen-doc.sh (hunk 4 above)
Comment (frankliee) [reply docs/gen-doc.sh]: Fixed

**Review by duanmeng — APPROVED** [review]
LGTM

**Pull request merged**